python/runtime/utils/PyRestRemoteClient.cpp

```diff
@@ -405,7 +405,7 @@ class PyNvcfRuntimeClient : public PyRestRemoteClient {
     });
 
     // Upload this request as an NVCF asset if needed.
-    if (request.code.size() > MAX_SIZE_BYTES) {
+    if (json(request).dump().size() > MAX_SIZE_BYTES) {
```

Comment:
Is there any way to just allocate a size padding to add to `request.code.size()` rather than construct a new, temporary, full JSON object copy just capture the resulting size?

Reply:
We could, but I think it might be a bit fragile. Do you have a recommendation for a reasonable size padding?
There was a discussion about possibly sending on the arguments to the server. In that case, we might need to revise the padding size or this check condition as a whole.

Another potential solution is to make it a 2-layer condition:  ` if (request.code.size() > MAX_SIZE_BYTES || json(request).dump().size() > MAX_SIZE_BYTES)`:  i.e., if the IR payload (the major one) is already large, then no need to check any further, hence saving the temp json allocation.

Reply:
I like the idea of the 2-layer condition. It prevents the worst case scenario (constructing large objects only to throw them away later), and I suppose the less-than-worst-case scenario probably isn't that big of a deal.

Reply:
I've revised this PR to implement the above strategy.